src/x11/x.c

```diff
@@ -404,7 +404,20 @@ static colored_layout *r_init_shared(cairo_t *c, notification *n)
         cl->l = create_layout(c);
 
         if (!settings.word_wrap) {
-                pango_layout_set_ellipsize(cl->l, PANGO_ELLIPSIZE_MIDDLE);
+                PangoEllipsizeMode ellipsize;
+                switch (settings.ellipsize) {
+                        case start:
+                                ellipsize = PANGO_ELLIPSIZE_START;
+                                break;
+                        case middle:
+                        default:
```

Comment:
Default should be an error condition, if ellipsize is anything other than `start`, `middle` or `end` it's most definitely a bug.

Reply:
Ok. In fact I copy/paste how the alignment is handle (with `left`, `center` and `right` values). So I don't know how you want me to handle this case.
Should I throw an error saying something went wrong and exiting?

Reply:
> So I don't know how you want me to handle this case.
Should I throw an error saying something went wrong and exiting?

An `assert` statement like should be all that's needed. It's not expected for the user to ever see that error so there's no need to be fancy with log messages and since there is specific error check here, `assert(false)` should suffice.